lib/loadObj.js

```diff
@@ -215,7 +215,7 @@ function loadObj(objPath, options) {
         }
 
         // UVs
-        if (defined(u)) {
+        if (defined(u) && (globalUvs.length > 0)) {
```

Comment:
So is an OBJ with normals that are referenced by not defined a valid OBJ? I can see it loads fine in Blender. I'm just wondering how far does this need to go to validate the file.

For example, if you add just one normal definition, this check passes, but most of the normals are still undefined, and you get the same problem again. We could check the length of globalNormals compared to the index it's asking for, but I'm not sure how common that is vs an OBJ with just 0 defined normals?

Reply:
It falls under the case of "probably not valid but easy to work around", which I think is an ok mentality for an obj reader (not for more well defined formats).

If some of the normals are defined, but not all, that's a harder issue to work around. I added some runtime errors when that happens.